src/index.js

```diff
@@ -129,14 +162,18 @@ let Rosnodejs = {
     const nodeOpts = options.node || {};
     rosNode = new RosNode(nodeName, rosMasterUri, nodeOpts);
 
-    return this._loadOnTheFlyMessages(options)
-      .then(_checkMasterHelper)
+    return new Promise((resolve,reject)=>{
```

Comment:
does explicitly creating a new promise here do something that I'm missing?

Reply:
We were previously not rejecting when an error occurred during initialization (and my timeout logic wouldn't work), I added this so I could add an explicit rejection in the catch block below.